lib/fizzy/execute.cpp

```diff
@@ -716,7 +718,8 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, span<const Value>
             const auto expected_type_idx = read<uint32_t>(immediates);
             assert(expected_type_idx < instance.module.typesec.size());
 
-            const auto elem_idx = stack.pop();
+            // TODO: is this the correct stack type?
+            const auto elem_idx = stack.pop().as<uint64_t>();
```

Comment:
@gumb0 this here is elem_idx type for `call_indirect`

Reply:
This one is also `i32` https://webassembly.github.io/spec/core/valid/instructions.html#xref-syntax-instructions-syntax-instr-control-mathsf-call-indirect-x